pkg/synthetictests/disruption_test.go

```diff
@@ -0,0 +1,120 @@
+package synthetictests
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_dnsOverlapDisruption(t *testing.T) {
+	now := time.Now()
+
+	type args struct {
+		events monitorapi.Intervals
+	}
+	tests := []struct {
+		name string
+		args args
+		want []*junitapi.JUnitTestCase
+	}{
+		{
+			name: "Detect disruption just after DNS lookup errors",
+			args: args{
+				events: []monitorapi.EventInterval{
+					{
+						Condition: monitorapi.Condition{
+							Locator: "disruption/openshift-api connection/new",
+							Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began for disruption/openshift-api connection/new requests over new connections: Get \"https://api.ci-op-v352gg5m-99751.aws-2.ci.openshift.org:6443/apis/image.openshift.io/v1/namespaces/default/imagestreams\": dial tcp: lookup api.ci-op-v352gg5m-99751.aws-2.ci.openshift.org: i/o timeout (likely a problem in cluster running tests, not the cluster under test)",
```

Comment:
I would extract two constants for Messages, one for DNS disruption, and one for real, will help with readability.